pkg/util/azureclient/mgmt/authorization/roleassignments.go

```diff
@@ -13,6 +13,8 @@ import (
 // RoleAssignmentsClient is a minimal interface for azure RoleAssignmentsClient
 type RoleAssignmentsClient interface {
 	Create(ctx context.Context, scope string, roleAssignmentName string, parameters mgmtauthorization.RoleAssignmentCreateParameters) (result mgmtauthorization.RoleAssignment, err error)
+	DeleteByID(ctx context.Context, roleID string) (result mgmtauthorization.RoleAssignment, err error)
+	List(ctx context.Context, filter string) (result mgmtauthorization.RoleAssignmentListResultPage, err error)
```

Comment:
please move List to a RoleAssignmentsClientAddons and have it return a slice of RoleAssignments.  Rationale: standardise with the rest of our clients